server/jetstream_cluster.go

```diff
@@ -8379,7 +8380,7 @@ RETRY:
 	releaseSyncOutSem()
 
 	if n.GroupLeader() == _EMPTY_ {
-		return fmt.Errorf("catchup for stream '%s > %s' aborted, no leader", mset.account(), mset.name())
+		return errors.Join(errCatchupAbortedNoLeader, fmt.Errorf("catchup for stream '%s > %s' aborted, no leader", mset.account(), mset.name()))
```

Comment:
Not sure if there's a more desirable version available in Go, I want to keep the more detailed error message containing account+stream names, but need to check it's of a specific error type also.

Reply:
this would be an option or something like: 
`fmt.Errorf("%w: catchup for stream '%s > %s' aborted, no leader", errCatchupAbortedNoLeader , mset.account(), mset.name())`

Reply:
to try to keep the same output you can refactor things a bit and do something as follows:

```go
package main

import (
	"errors"
	"fmt"
)

type catchupErr struct {
	msg string
	err error
}

func (e *catchupErr) Error() string {
	return e.msg
}

func (e *catchupErr) Unwrap() error {
	return e.err
}

// any other catchup error
var errCatchupAbortedNoLeader = errors.New("catchup failed, no leader")

func main() {
	account := "foo"
	name := "bar"
	err := &catchupErr{
		msg: fmt.Sprintf("catchup for stream '%s > %s' aborted, no leader", account, name),
		err: errCatchupAbortedNoLeader,
	}
	fmt.Println("--->", err)
	fmt.Println("result: ", errors.Is(err, errCatchupAbortedNoLeader))
        // ---> catchup for stream 'foo > bar' aborted, no leader
        // result:  true
}
```

Reply:
Done, great suggestion, thanks @wallyqs !